molr-mole-core/src/main/java/io/molr/mole/core/single/SingleNodeMissionExecutor.java

```diff
@@ -46,20 +48,25 @@
 
     private final ExecutorService executorService;
 
+    private final Appender appender;
 
-    public SingleNodeMissionExecutor(SingleNodeMission<R> mission, Map<String, Object> parameters) {
+
+    public SingleNodeMissionExecutor(MissionHandle handle, SingleNodeMission<R> mission, Map<String, Object> parameters) {
+        this.handle = handle;
         this.mission = requireNonNull(mission, "mission must not be null");
         executorService = Executors.newSingleThreadExecutor();
         this.input = MissionInput.from(parameters);
         this.representation = SingleNodeMissions.representationFor(mission);
         this.representations.onNext(this.representation);
         this.output = new BlockOutputCollector(outputCollector, representation.rootBlock());
+        this.appender = new MissionLogAppender(handle);
+        LOGGER.addAppender(appender);
         publishState();
     }
 
     private void resume() {
         if (started.getAndSet(true)) {
-            LOGGER.warn("Already Running. Doing nothing.");
+            LOGGER.warn(MissionLog.from(handle).strand(singleStrand).message("Already Running. Doing nothing.").build());
```

Comment:
I am a bit worried about those builders ... was there any attempt to try it without them? The point is: If we need them, then any java mole has to use them ... we get nothing automatic....

Reply:
In other words: Is there no way to inject the handle and the strandId into the logger otherwise?

Reply:
the main point being: If there is no way without hose builders, then we loose any log information from 3rd party libs, right?